wikihow2zim/scraper.py

```diff
@@ -29,9 +38,73 @@ def __init__(self, **kwargs):
         for option in options:
             setattr(self, option, kwargs[option])
 
+        if self.output_dir is None:
+            self.output_dir = pathlib.Path(os.getcwd()).joinpath("output")
+
+        pathlib.Path(self.output_dir).mkdir(parents=True, exist_ok=True)
+
         self.url = URLS[self.language]
 
+        # jinja2 environment setup
+        self.env = Environment(
+            loader=FileSystemLoader(str(self.templates_dir)), autoescape=True
+        )
+
+        # Set up the output zim file:
+        fpath = pathlib.Path(self.output_dir).joinpath("output.zim")
+        self.creator = Creator(filename=fpath).set_mainpath("Home")
+        self.creator.start()
+
+    @property
+    def templates_dir(self):
+        return ROOT_DIR.joinpath("templates")
+
+    def add_assets(self):
+        assets_path = ROOT_DIR.joinpath("assets")
+        for root, dirs, files in os.walk(assets_path):
+            for file in files:
+                fpath = pathlib.Path(os.path.join(root, file))
+                path = os.path.relpath(fpath, start=ROOT_DIR)
+                self.creator.add_item_for(path=path, fpath=fpath)
+
+    def add_homepage(self):
+        template = self.env.get_template("base.html")
+        self.creator.add_item_for(
+            path="Home",
+            title="Home",
+            content=template.render(title="Home", description="Home"),
+            mimetype="text/html",
+        )
+
+        # Add wikihow logo:
+        LOGO_PATH = "https://www.wikihow.com/extensions/wikihow/mobile/images/wikihow_logo_230.png"
+        logo_path = ROOT_DIR.joinpath("assets", "static", "wikihow_logo.png")
+        self.creator.add_item(
+            URLItem(url=LOGO_PATH, path="logo_path", mimetype="image/png")
+        )
+
+        # fetch icon from website and set it as Zim Icon
```

Comment:
All of the above is related to the illustration, not the homepage. That should be moved.